src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx

```diff
@@ -31,26 +31,38 @@ import {
 import { PhrasalTemplatePhraseProps } from './PhrasalTemplatePhrase.types';
 import { getFieldPlaceholders, getNewDefaultField } from '../PhrasalTemplate.utils';
 import { KEYWORDS } from '../PhrasalTemplateField/PhrasalTemplateField.const';
+import { PhrasalTemplateResponseValuePhraseSchema } from '../../../../../../pages/BuilderApplet/BuilderApplet.schema';
 
 export const PhrasalTemplatePhrase = ({
   name = '',
   responseOptions = [],
   index = 0,
   onRemovePhrase,
+  field,
 }: PhrasalTemplatePhraseProps) => {
   const { t } = useTranslation('app');
+  // Use `useEffect` to trigger validation when `fields` or form state changes
   const [removePopupOpen, setRemovePopupOpen] = useState(false);
   const [previewPopupOpen, setPreviewPopupOpen] = useState(false);
-  const [addItemMenuAnchorEl, setAddItemMenuAnchorEl] = useState<null | HTMLElement>(null);
+  const [previewPhraseDisabled, setPreviewPhraseDisabled] = useState(false);
 
+  const [addItemMenuAnchorEl, setAddItemMenuAnchorEl] = useState<null | HTMLElement>(null);
   const phraseFieldsName = `${name}.fields`;
-  const { control, setValue, getValues, formState } = useCustomFormContext();
-
+  const { control, setValue, getValues, formState, watch } = useCustomFormContext();
   const { fields, append, remove, swap } = useFieldArray({
     control: control as unknown as Control<{ values: TPhrasalTemplateField[] }>,
     name: phraseFieldsName as 'values',
   });
 
+  useEffect(() => {
+    const validateField = async () => {
+      const result = await validatePhraseField(field);
+      setPreviewPhraseDisabled(result);
+    };
+
+    validateField(); // Call the async function
+  }, [field, formState, formState.isSubmitting]);
```

Comment:
Nit: I think it's redundant to include both `formState` and `formState.isSubmitting` as deps here; changes to `formState.isSubmitting` is implied by changes to `formState`.
```suggestion
  }, [field, formState]);
```

Reply:
Looks like this one still is unresolved and wasn't made as part of @AlejandroCoronadoN last [commit](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1977/commits/f6a19795dd1606b98a7498bfa4033abaabb9456c)

Reply:
I've just taken care of this :+1: